### DISCUSSION TITLE
NAT Service not working outside of local network.

### DISCUSSION THREAD

**Discussion by ashermyers** [amidaware/tacticalrmm#1125] [General]
The port 4222 tcp is not working outside of my local network. Telnet works locally but outside of the network does not work... IP tables are currently accepting incoming and outgoing connections and ufw is disabled... Don't know what the issue is.

**Comment by wh1te909** [comment]
you need to port forward 4222 and 443 in your router to the internal IP of the vm

**Reply by ashermyers** [reply]
I'm using google cloud with all ports open

**Reply by wh1te909** [reply]
paste output of `sudo systemctl status nats`

**Reply by ashermyers** [reply]
I'm on docker.

**Reply by wh1te909** [reply]
docker is more for dev/testing locally. if you are on google cloud, use standard install with a VM

**Comment by ashermyers** [comment]
Also, I am using google cloud with https and http connections enabled... And the dashboard works and the nat service is running but non accessible outside my network

**Reply by silversword411** [reply]
I haven't used Google cloud vm's much and docker less, but you have your firewall/environment is setup wrong and 4222 isn't exposed to the internet for TRMM clients to connect to. 

Read google cloud docs and figure out how to get your docker to expose the nats port publicly on port 4222.

**Comment by ninjamonkey198206** [comment]
If you're using Google cloud, you also have to open the port in the gcp firewall rules, not just in the VM. I had this working in a gcp VM with docker and HAProxy, so it's perfectly capable.

If you need help finding the settings, let me know.
